src/main/java/org/opentripplanner/model/TripPattern.java

```diff
@@ -161,21 +161,47 @@ public void setHopGeometriesFromPattern(TripPattern other) {
         int sizeOfShortestPattern = Math.min(numberOfStops(), other.numberOfStops());
 
         for (int i = 0; i < sizeOfShortestPattern - 1; i++) {
-            if (other.getHopGeometry(i) != null
-                && other.getStop(i).equals(this.getStop(i))
-                && other.getStop(i + 1).equals(this.getStop(i + 1))) {
+
+            LineString hopGeometry = other.getHopGeometry(i);
+            if (hopGeometry == null) {
+                // Create new straight-line geometry for hop
+                this.setHopGeometry(
+                        i,
+                        GeometryUtils.getGeometryFactory().createLineString(
+                                new Coordinate[]{
+                                        coordinate(stopPattern.getStop(i)),
+                                        coordinate(stopPattern.getStop(i + 1))
+                                }
+                        )
+                );
+                continue;
+            }
+
+            if (other.getStop(i).equals(this.getStop(i)) && other.getStop(i + 1)
+                    .equals(this.getStop(i + 1))) {
                 // Copy hop geometry from previous pattern
                 this.setHopGeometry(i, other.getHopGeometry(i));
-            } else {
-                // Create new straight-line geometry for hop
-                this.setHopGeometry(i,
-                    GeometryUtils.getGeometryFactory().createLineString(
-                        new Coordinate[]{
-                            coordinate(stopPattern.getStop(i)),
-                            coordinate(stopPattern.getStop(i + 1))
-                        }
-                    )
+            }
+            else {
+                // Use old geometry but patch first and last point with new stops
+                Coordinate newStart;
```

Comment:
Is there a reason that you put variable definition and assignment on two lines?

Reply:
No reason at all. I joined definition and assignment into one statement now.